crates/ruff_formatter/src/buffer.rs

```diff
@@ -464,6 +502,68 @@ impl<Context> Buffer for RemoveSoftLinesBuffer<'_, Context> {
     }
 }
 
+#[derive(Copy, Clone, Debug, Default)]
+enum RemoveSoftLineBreaksState {
+    #[default]
+    Default,
+    InIfGroupBreaks {
+        conditional_content_level: usize,
+    },
+}
+
+impl RemoveSoftLineBreaksState {
+    fn should_drop(&mut self, element: &FormatElement) -> bool {
+        match self {
+            Self::Default => false,
+            Self::InIfGroupBreaks { .. } => {
+                match element {
+                    FormatElement::Tag(Tag::StartConditionalContent(_)) => {
+                        self.increment_conditional_content();
+                    }
+                    FormatElement::Tag(Tag::EndConditionalContent) => {
+                        self.decrement_conditional_content();
+                    }
+                    _ => {}
+                }
+
+                true
+            }
+        }
+    }
+
+    fn decrement_conditional_content(&mut self) -> bool {
```

Comment:
I don't think the return type of this method is being used anywhere, we should remove it.